src/com/github/mtdrewski/GRAPH_moment/controller/GraphDrawerController.java

```diff
@@ -107,27 +108,22 @@ protected void exitEdgeMode(boolean success) {
         inEdgeMode = false;
     }
 
-    public void passGraphAndDraw(Graph graphFromInput) {
-        graph = graphFromInput;
-        reDrawGraph();
-    }
-
-    private void reDrawGraph() {
+    public void reDrawGraph(Graph newGraph) {
 
         anchorPane.getChildren().clear();
+        graph = new Graph();
         ArrayList<VertexCircle> vertexCircles = new ArrayList<>();
```

Comment:
Removing passGraphAndDraw can be removen but I think the name reDrawGraph should be changed since we don't redraw a graph, but draw a new one.